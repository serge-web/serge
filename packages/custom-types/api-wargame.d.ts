import { PlayerUiDispatch, PlayerUiActionTypes } from './player-ui-actions'
import Message from './message.d'

export type ApiWargameDb = PouchDB.Database<Message>

export interface ApiWargameListenNewMessage {
  db: ApiWargameDb,
  name: string,
  dispatch: PlayerUiDispatch
}

export type ApiWargameDbObject = { db: ApiWargameDb, name: string }
