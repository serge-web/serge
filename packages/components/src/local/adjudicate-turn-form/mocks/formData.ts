import { AdjudicateTurnFormPopulate, AttributeTypes } from '@serge/custom-types'

const attributes: AttributeTypes = []

const formData: AdjudicateTurnFormPopulate = {
  contactId: 'C234',
  status: [
    {
      name: 'Fishing',
      mobile: false
    },
    {
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }
  ],
  speed: [10, 20, 30],
  visibleTo: [
    {
      name: 'Blue Force',
      colour: '#69c',
      id: 'Blue'
    },
    {
      name: 'Red Force',
      colour: '#f00',
      id: 'Red'
    },
    {
      name: 'White Force',
      colour: '#fff',
      id: 'umpire'
    }
  ],
  condition: ['Working', 'Disabled', 'Immobile', 'Destroyed'],
  attributes: attributes
}

export default formData
