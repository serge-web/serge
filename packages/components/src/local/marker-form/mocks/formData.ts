import { PerceptionFormData } from '@serge/custom-types'

const formData: PerceptionFormData = {
  populate: {
    perceivedForces: [
      {
        name: 'Blue Force',
        colour: '#69c',
        id: 'Blue'
      },
      {
        name: 'Red Force',
        colour: '#f00',
        id: 'Red'
      },
      {
        name: 'White Force',
        colour: '#fff',
        id: 'Umpire'
      }
    ],
    perceivedTypes: [
      { uniqid: 'aa', name: 'Fishing vessel' },
      { uniqid: 'a1', name: 'torpedo' },
      { uniqid: 'a2', name: 'Boghammer' },
      { uniqid: 'a3', name: 'Frigate' }
    ]
  },
  values: {
    perceivedNameVal: 'Test',
    perceivedForceId: 'Blue Force',
    perceivedForceColor: '#69c',
    perceivedTypeId: 'a3',
    assetId: 'c00012',
    iconURL: 'frigate.svg'
  }
}

export default formData
