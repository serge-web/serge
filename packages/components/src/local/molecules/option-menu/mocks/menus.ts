import { faBath, faClone, faFileDownload, faTrash } from '@fortawesome/free-solid-svg-icons'

export default [
  {
    onClick: (): void => window.alert('Click on menu #1'),
    icon: faBath,
    title: 'Save clean copy of wargame (no messages)',
    label: 'Clean copy'
  },
  {
    onClick: (): void => window.alert('Click on menu #2'),
    icon: faClone,
    title: 'Duplicate wargame',
    label: 'Duplicate'
  },
  {
    onClick: (): void => window.alert('Click on menu #3'),
    icon: faFileDownload,
    title: 'Export wargame',
    label: 'Export'
  },
  {
    onClick: (): void => window.alert('Click on menu #4'),
    icon: faTrash,
    title: 'Delete wargame',
    label: 'Delete'
  }
]
