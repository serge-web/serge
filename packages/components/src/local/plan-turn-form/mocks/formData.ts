const formData = {
  populate: {
    status: [{
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }],
    speed: [10, 20, 30]
  },
  values: {
    statusVal: {
      name: 'Transiting',
      mobile: true
    },
    speedVal: 10,
    turnsVal: 5,
    condition: 'Working'
  }
}

export default formData
