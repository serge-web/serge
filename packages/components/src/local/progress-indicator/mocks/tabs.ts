export const tabs = [
  {
    name: 'overview',
    complete: false
  },
  {
    name: 'forces',
    complete: false
  },
  {
    name: 'channels',
    complete: false
  }
]

export const tabsVariant = [
  {
    name: 'overview',
    complete: true
  },
  {
    name: 'forces',
    complete: false
  },
  {
    name: 'channels',
    complete: false
  }
]

export default tabs
