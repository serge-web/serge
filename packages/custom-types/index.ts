/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as Asset } from './asset.d'
export * from './attributes'
export * from './channel-data.d'
export { default as ChannelUI } from './channel-ui.d'
export { default as Channel } from './channel.d'
export { default as Editor } from './editor.d'
export * from './export-items-actions.d'
export * from './export-items.d'
export { default as ForceData } from './force-data.d'
export { default as ForceOption } from './force-option.d'
export { default as ForceRole } from './force-role.d'
export { default as Force } from './force.d'
export * from './forms.d'
export * from './global.d'
export { default as GroupItem } from './group-item.d'
export { default as IconOption } from './icon-option.d'
export * from './map-annotation'
export { default as MapPostBack } from './map-postback-data.d'
export { default as MappingConstraints } from './mapping-constraints.d'
export { default as MappingContext } from './mapping-context.d'
export { default as MarkerPerception } from './marker-perception.d'
export * from './message-types-actions.d'
export { default as MessageTypes, TemplateBodysByKey } from './message-types.d'
export {
  ChatMessage, CollaborationDetails, CoreMessage, default as Message, FeedbackItem, InteractionDetails
  // All template based messages
  , InteractionMessageStructure, MessageChannel, MessageCloneMarker, MessageCounter, MessageCreateTaskGroup,
  // All template based messages
  MessageCustom, MessageDeleteMarker, MessageDeletePlatform, MessageDetails,
  MessageDetailsFrom,
  // System Messages
  MessageFeedback,
  // Map Types
  MessageForceLaydown, MessageHostPlatform, MessageInfoType,
  MessageInfoTypeClipped, MessageInteraction, MessageLeaveTaskGroup, MessageMap, MessagePerceptionOfContact, MessagePlanning,
  MessageStateOfWorld, MessageStructure, MessageSubmitPlans, MessageUpdateMarker, MessageVisibilityChanges, PlanningMessageStructure
} from './message.d'
export { MessagesActionTypes, MessagesDispatch, SetCurrentViewFromURI } from './messages-action'
export * from './participant.d'
export { default as PerceivedTypes } from './perceived-types.d'
export { default as Perception } from './perception.d'
export { default as PlanMobileAsset, TurningDetails } from './plan-mobile-asset'
export { default as PlannedRoute } from './planned-route.d'
export * from './planning-activity'
export { default as PlatformTypeData } from './platform-type-data.d'
export { default as PlatformType } from './platform-type.d'
export * from './player-log-actions'
export * from './player-log.d'
export * from './player-ui-actions.d'
export { default as PlayerUi, PlayerUiChannels, PlayerUiChatChannel } from './player-ui.d'
export { default as RequestForInformation } from './request-for-information.d'
export { default as Role } from './role.d'
export * from './root-state'
export { default as RouteMarker, RouteMarkerDuo, RouteMarkerStatus as RouteMarkerStatus } from './route-marker'
export { default as RouteStatus } from './route-status.d'
export { default as RouteStore } from './route-store.d'
export { default as RouteTurn } from './route-turn'
export { default as Route, RouteChild } from './route.d'
export { default as SelectedAsset } from './selected-asset.d'
export * from './serge-hex-3.d'
export { default as SetWargameMessage } from './set-wargame-message.d'
export { AssetState, ForceState, StateOfWorld } from './state-of-world.d'
export { default as State } from './state.d'
export { default as Template, TemplateBody } from './template.d'
export * from './turn-length.d'
export { default as Visibility } from './visibility.d'
export { WargameActionTypes, WargameDispatch, WargameRevision } from './wargame-action'
export { default as WargameData } from './wargame-data.d'
export { default as WargameList } from './wargame-list.d'
export { default as WargameOverview } from './wargame-overview.d'
export { default as Wargame } from './wargame.d'

