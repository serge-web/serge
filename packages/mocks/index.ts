/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as GameMessagesMock } from './game-messages.mock'
export { default as GameMessagesMockRFI } from './game-messages-rfi.mock'
export { default as AdminMessagesMock } from './admin-messages.mock'
export { default as ChatMessagesMock } from './chat-messages.mock'
export { default as infoMarker } from './map-annotation.mock'
export { default as InfoMessagesMock } from './info-messages.mock'
export { default as annotationMarkerData } from './annotation-marker-data.mosk'
export { default as GameChannels2 } from './game-channels-2.mock'
export { default as WargameMock } from './wargame.mock'
export { default as MapAnnotationMock } from './map-annotation.mock'
export { default as WargameExportedMock } from './wargame-exported.mock'
export { default as P9Mock } from './p9.mock'
export { default as MessageTemplatesMock } from './message-templates.mock'
export { default as MessageTemplatesMockByKey } from './message-templates-by-key.mock'
export { default as localMappingConstraints } from './local-mapping.mock'
export { default as forces } from './forces.mock'
export { default as atlanticForces } from './atlantic-forces.mock'
export { default as smallForces } from './small-forces.mock'
export { default as smallScaleForces } from './small-scale-forces.mock'
export { default as watuWargame } from './watu-wargame-mock'
export { default as watuPlaytest } from './watu-playtest-mock'
export { default as p9wargame } from './p9.mock'
export { default as cmdWkWargame } from './cmd-week.mock'
export { default as smallScaleMappingConstraints } from './small-scale-mapping-constraints.mock'
export { default as preInitialisedForces } from './pre-initialised-forces.mock'
export { default as platformTypes } from './platform-types.mock'
export { default as platformType } from './platform-type.mock'
export { default as selectedAsset } from './selected-asset.mock'
export { default as adminTabs } from './admin-tabs.mock'
export { default as feedbackMessage, feedbackMessage1, feedbackMessage2 } from './feedback-message.mock'
export { default as actionFeedbackMessages } from './action-feedback-messages.mock'
export { default as actionSetLatestFeedbackMessage } from './action-set-latest-feedback-message.mock'
export { default as actionSetLatestWargameMessage } from './action-set-latest-wargame-message.mock'
export { default as actionSetWargameMessages } from './action-set-wargame-messages.mock'
export { default as actionOpenMessage } from './action-open-message.mock'
export { default as actionCloseMessage } from './action-close-message.mock'
export * from './coa-channel-data-2.mock'
export * from './planning-messages.mock'
export { default as planningMessageTemplatesMock } from './planning-message-templates.mock'
export * from './coa-messages.mock'
