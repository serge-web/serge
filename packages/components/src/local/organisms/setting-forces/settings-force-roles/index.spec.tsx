import React from 'react'
import renderer from 'react-test-renderer'
import { forces } from '@serge/mocks'
import { ForceData } from '@serge/custom-types'
import RolesAccordion from './index'

const handleChange = (obj: ForceData): void => {
  console.log(obj)
}

describe('RolesAccordion component:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <RolesAccordion
          data={forces[0]}
          handleChangeForce={handleChange}
          forces={forces}
        />,
        { createNodeMock: () => document.createElement('textarea') }
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
