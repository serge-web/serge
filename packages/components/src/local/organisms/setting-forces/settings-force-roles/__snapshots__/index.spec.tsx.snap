// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RolesAccordion component: renders correctly 1`] = `
<div
  className="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root accordion-header"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root accordion-title MuiTypography-body1"
      >
        Roles
      </p>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          className="svg-inline--fa fa-caret-down"
        />
      </span>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiAccordionDetails-root"
          >
            <div
              className="row"
            >
              <div
                className="col section"
              >
                <div
                  className="main"
                >
                  <div
                    className="placeholder"
                  >
                    Roles
                  </div>
                  <div
                    className="content"
                  >
                    <div
                      className="main"
                    >
                      <ul
                        className="main"
                      >
                        <li
                          className="li"
                        >
                          <div
                            className="presection"
                          >
                            <section
                              className="section"
                            >
                              <div
                                className="value-label"
                              >
                                <div
                                  onClick={[Function]}
                                >
                                  Game Control
                                </div>
                              </div>
                              <span>
                                <div
                                  className="trash-icon"
                                  onClick={[Function]}
                                >
                                  <svg
                                    className="svg-inline--fa fa-trash"
                                  />
                                </div>
                              </span>
                              <div
                                className="role"
                              >
                                <div
                                  className="role-item"
                                >
                                  <div
                                    className="main"
                                    title="Click to Edit"
                                  >
                                    <section
                                      className="password-mask"
                                    >
                                      <div
                                        className="dots"
                                      >
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                      </div>
                                    </section>
                                    <div
                                      className="key"
                                      onClick={[Function]}
                                    >
                                      <svg
                                        className="svg-inline--fa fa-key"
                                      />
                                    </div>
                                  </div>
                                  <div
                                    className="role-title"
                                  >
                                    Password
                                  </div>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                  <div
                                    className="role-title title-center"
                                    title="Game Control"
                                  >
                                    <svg
                                      className="svg-inline--fa fa-chess-king"
                                    />
                                  </div>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                  <div
                                    className="role-title title-center"
                                    title="Can view all channels"
                                  >
                                    <svg
                                      className="svg-inline--fa fa-eye"
                                    />
                                  </div>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                  <div
                                    className="role-title title-center"
                                    title="Can view feedback/insights"
                                  >
                                    <svg
                                      className="svg-inline--fa fa-comments"
                                    />
                                  </div>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                  <div
                                    className="role-title title-center"
                                    title="Can release RFI responses"
                                  >
                                    <svg
                                      className="svg-inline--fa fa-book-open-reader"
                                    />
                                  </div>
                                </div>
                              </div>
                              <span />
                            </section>
                          </div>
                        </li>
                        <li
                          className="li"
                        >
                          <div
                            className="presection"
                          >
                            <section
                              className="section"
                            >
                              <div
                                className="value-label"
                              >
                                <div
                                  onClick={[Function]}
                                >
                                  RFI Manager
                                </div>
                              </div>
                              <span>
                                <div
                                  className="trash-icon"
                                  onClick={[Function]}
                                >
                                  <svg
                                    className="svg-inline--fa fa-trash"
                                  />
                                </div>
                              </span>
                              <div
                                className="role"
                              >
                                <div
                                  className="role-item"
                                >
                                  <div
                                    className="main"
                                    title="Click to Edit"
                                  >
                                    <section
                                      className="password-mask"
                                    >
                                      <div
                                        className="dots"
                                      >
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                        <svg
                                          className="svg-inline--fa fa-circle"
                                        />
                                      </div>
                                    </section>
                                    <div
                                      className="key"
                                      onClick={[Function]}
                                    >
                                      <svg
                                        className="svg-inline--fa fa-key"
                                      />
                                    </div>
                                  </div>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                </div>
                                <div
                                  className="role-item"
                                >
                                  <span
                                    className="MuiSwitch-root MuiSwitch-sizeSmall"
                                  >
                                    <span
                                      aria-disabled={false}
                                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
                                      onBlur={[Function]}
                                      onDragLeave={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex={null}
                                    >
                                      <span
                                        className="MuiIconButton-label"
                                      >
                                        <input
                                          checked={true}
                                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                                          disabled={false}
                                          onChange={[Function]}
                                          type="checkbox"
                                        />
                                        <span
                                          className="MuiSwitch-thumb"
                                        />
                                      </span>
                                    </span>
                                    <span
                                      className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
                                    />
                                  </span>
                                </div>
                              </div>
                              <span />
                            </section>
                          </div>
                        </li>
                      </ul>
                      <div
                        className="add-section"
                      >
                        <button
                          className="button"
                          onClick={[Function]}
                        >
                          Add Role
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
